KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,348 +1,656 @@
-            entire14,
-            references14,
-            paras,
-            addendum,
-            ref14II,
-            ref14III,
-            specref14))
-
-"""
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-If you are writing something that will make a change to the chapter files, write it BEFORE this line, this part
-is where the lists representing the words/strings in the chapter are joined together and updated as a string!
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""
-
-
-# write to files
-# new output files for safety
-with open("updated9.tex", "w") as temp9:
-    temp9.write(str9)
-
-with open("updated14.tex", "w") as temp14:
-    temp14.write(str14)
+
+def main():
+    """
+    Runs all of the other functions and outputs their results into an output file as well as putting together the list
+    of additions that is fed into fix_chapter.
+
+    """
+    chap_nums = []
+    new_commands = []  # used to hold the indexes of the commands
+
+    # contains the locations of things with "paragraph{" in KLSadd, these are subsections that need to be sorted
+    klsaddparas = []
+    math_people = []
+
+    # Stores all keywords
+    kls_list_full = []
+
+    sortmatch_2 = []
+
+    # open the KLSadd file to do things with
+    with open(DATA_DIR + "KLSaddII.tex", "r") as add:
+        # store the file as a string
+        addendum = add.readlines()
+        kls_list_all = new_keywords(addendum, kls_list_full)[0]
+        # Makes sections look like other sections
+        sorter_check = new_keywords(addendum, kls_list_full)[1]
+
+        for item in addendum:
+            addendum[addendum.index(item)] = item.replace("\\eqref{","\\eqref{KLS Addendum: ")
+
+        for word in addendum:
+            if "paragraph{" in word:
+                lenword = len(word) - 1
+                temp = word[0:word.find("{") + 1] + "\\bf KLS Addendum: " + word[word.find("{") + 1: lenword]
+                addendum[addendum.index(word)] = temp
+            if "subsubsection*{" in word:
+                lenword = len(word) - 1
+                addendum[addendum.index(word)] = word[0:word.find("{") + 1] + "\\bf KLS Addendum: " + \
+                word[word.find("{") + 1: lenword]
+        index = 0
+        indexes = []
+
+        # Designates sections that need stuff added
+        # gets the index
+        startindex = 99999
+        #number must be larger than all the others, thus 99999
+        for word in addendum:
+            index += 1
+            if "." in word and "\\subsection*{" in word:
+                if "9." in word:
+                    chap_nums.append(9)
+                    name = word[word.find("{") + 1: word.find("}")]
+                    math_people.append(name + "#")
+                if "14." in word:
+                    chap_nums.append(14)
+                    name = word[word.find("{") + 1: word.find("}")]
+                    math_people.append(name + "#")
```

Comment:
These conditions result in the same thing but the chap_nums part - the rest should not be in the if